src/blockrelay/graphene.cpp

```diff
@@ -536,7 +551,9 @@ std::set<uint64_t> CGrapheneBlock::UpdateResolvedTxsAndIdentifyMissing(std::map<
         const auto &elem = mapPartialTxHash.find(cheapHash);
         if (elem != mapPartialTxHash.end())
         {
-            vTxHashes256.push_back(elem->second);
+            const auto repeat = std::find(vTxHashes256.begin(), vTxHashes256.end(), elem->second);
```

Comment:
It seems like a red flag to me if there are repeat elements in `blockCheapHashes` because these come directly from the IBLT. And only unique elements should be added to the IBLT. So these repeats have been observed in testing?

Reply:
yes

Reply:
Right, this code should be here to filter out duplicates because any attacker could add them, but seems we have also have a problem upstream where duplicates are somehow being added.

Reply:
Here is some example output: 

```************************
EXCEPTION: St13runtime_error       
Cannot OrderTxHashes if size of vTxHashes256 unequal to nBlockTxs       
bitcoin in ProcessMessages()
andrewstone
8:56 PM - Fri
nBlockTxs is correct
vTxHashes256 is double
and repeats itself
$20 = std::vector of length 6, capacity 8 = {{<base_blob<256>> = {data = "u\314Ӵ0\254\356\v\177a\321\313t\251\003\235\204\321&\311c\237\320ؠ\211\261\004\336'\330j"}, <No data fields>}, {<base_blob<256>> = {data = "\205G\317O\345\240\070\271\336\274]\265\223\261\346C\266\261\244\302\371z\266\032\061\366Ș\031"}, <No data fields>}, {<base_blob<256>> = {data = "\022\002\222\t\230\310\026\226:\227\004,R\306\344\240\351\063\253\255x7\343\324O\033\246\\7(\271J"}, <No data fields>}, {<base_blob<256>> = {data = "\205G\317O\345\240\070\271\336\274]\265\223\261\346C\266\261\244\302\371z\266\032\061\366Ș\031"}, <No data fields>}, {<base_blob<256>> = {data = "u\314Ӵ0\254\356\v\177a\321\313t\251\003\235\204\321&\311c\237\320ؠ\211\261\004\336'\330j"}, <No data fields>}, {<base_blob<256>> = {data = "\022\002\222\t\230\310\026\226:\227\004,R\306\344\240\351\063\253\255x7\343\324O\033\246\\7(\271J"}, <No data fields>}}```